apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/SpanImpl.java

```diff
@@ -174,6 +181,16 @@ public SpanImpl withAction(@Nullable String action) {
         return this;
     }
 
+    @Override
+    public void addEndListener(SpanEndListener<? super SpanImpl> listener) {
+        endListeners.add(listener);
```

Comment:
We should probably add a warning here for the set size exceeding some threshold in case we generate a leak somehow. And drop adding if so